helpers/filterSeats.js

```diff
@@ -302,6 +302,62 @@ module.exports = function(dust) {
       reply = newReply;
     }
 
+    function getLeftCentreRightMergedSection(leftSection, centreSection, rightSection) {
+      var mergedLeftRightSection = getLeftRightMergedSection(leftSection, rightSection);
+      var leftRightRates = mergedLeftRightSection.rates;
+      var centreRates = centreSection.rates;
+      var centreColours = [];
+      var newRates = [];
+
+      _.forEach(centreRates, function(rate) {
+        centreColours.push(rate.links.ticketRates.colour);
+        newRates.push(rate);
+      });
+
+
+      _.forEach(leftRightRates, function(rate) {
+        var currColour = rate.links.ticketRates.colour;
+        if (centreColours.indexOf(currColour) === -1) {
+          newRates.push(rate);
+        }
+      });
+
+      newRates = _.orderBy(newRates, function(newRate) {
+        return newRate.grossPrice;
+      }, ['asc']);
+
+      centreSection.name = mergedLeftRightSection.name;
+      centreSection.rates = newRates;
+
+      return centreSection;
+    }
+
+    function mergeLeftCentreRight() {
+      var targetedSectionIndexes = [];
+
+      _.forEach(reply, function(section, index) {
+        if (section.name.toUpperCase().indexOf('CENTRE') !== -1) {
+          targetedSectionIndexes.push(index);
+        }
+      });
+
+      _.forEach(targetedSectionIndexes, function(centralSectionIndex) {
+        centralSectionIndex = parseInt(centralSectionIndex, 10);
+        var mergedSection = getLeftCentreRightMergedSection(
+          reply[centralSectionIndex - 1],
+          reply[centralSectionIndex],
+          reply[centralSectionIndex + 1]
+        );
+        reply[centralSectionIndex - 1].toBeRemoved = true;
+        reply[centralSectionIndex + 1].toBeRemoved = true;
+        reply[centralSectionIndex] = mergedSection;
+      });
+
+      reply = _.filter(reply, function(section) {
+        return !section.toBeRemoved;
+      });
+    }
+
```

Comment:
This whole section could do with comments and a flow to make it obvious what is happening